src/oemof/solph/_energy_system.py

```diff
@@ -32,12 +34,51 @@ class EnergySystem(es.EnergySystem):
     oemof.network directly.
     """
 
-    def __init__(self, **kwargs):
+    def __init__(
+        self, timeindex=None, timeincrement=None, timemode="implicit", **kwargs
+    ):
         # Doing imports at runtime is generally frowned upon, but should work
         # for now. See the TODO in :func:`constraint_grouping
         # <oemof.solph.groupings.constraint_grouping>` for more information.
         from oemof.solph import GROUPINGS
 
         kwargs["groupings"] = GROUPINGS + kwargs.get("groupings", [])
 
-        super().__init__(**kwargs)
+        if not (
+            isinstance(timeindex, pd.DatetimeIndex)
+            or isinstance(timeindex, type(None))
+        ):
+            msg = (
+                "Parameter 'timeindex' has to be of type "
+                "pandas.datetimeindex or NoneType and not of type {0}"
+            )
+            raise TypeError(msg.format(type(timeindex)))
+
+        self.timemode = timemode
+
+        if timemode == "implicit" and timeindex is not None:
+            # Add one timestep to the timeindex.
+            timeindex = timeindex.union(
+                pd.date_range(
+                    timeindex[-1] + timeindex.freq,
```

Comment:
I know, this is the way, it's currently done, as you also state:

> By now the first TIMEPOINT has been ignored. So all results do have the length of the number of intervals. This behaviour will be kept as default. For a storage the initial storage content can not be found in the results.

However, I consider this very confusing and would recommend changing this when the definitions are changed anyway.